inc/class-wpseo-content-images.php

```diff
@@ -29,11 +29,16 @@ public function get_images( $post_id, $post = null ) {
 	 *
 	 * @return array An array of image URLs.
 	 */
-	protected function get_images_from_content( $content ) {
+	public function get_images_from_content( $content ) {
+		if ( ! is_string( $content ) ) {
+			return array();
+		}
+
 		$content_images = $this->get_img_tags_from_content( $content );
 		$images         = array_map( array( $this, 'get_img_tag_source' ), $content_images );
 		$images         = array_filter( $images );
 		$images         = array_unique( $images );
+		$images         = array_values( $images );
```

Comment:
I'd consider to add an inline comment to explain why this is necessary (I didn't realize why at first glance, had look for the related commit).

Reply:
Done